data_diff/diff_tables.py

```diff
@@ -71,6 +73,13 @@ def __post_init__(self):
         if self.min_update is not None and self.max_update is not None and self.min_update >= self.max_update:
             raise ValueError("Error: min_update expected to be smaller than max_update!")
 
+    def with_schema(self) -> "TableSegment":
+        "Queries the table schema from the database, and returns a new instance of TableSegmentWithSchema."
```

Comment:
It's not actually a different class though, no?

Why does this instantiate a new segment when e.g. the count and checksum don't?

Reply:
Fair